analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -47,6 +46,313 @@ public void InstanceReference_SetsNotNull_VB()
         validator.TagValue("Me").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
     }
 
+    [DataTestMethod]
+    [DataRow("Where", "x => true")]
+    [DataRow("Select", "x => x")]
+    [DataRow("SelectMany", "x => new[] {1}")]
+    [DataRow("Take", "1")]
+    [DataRow("TakeWhile", "x => true")]
+    [DataRow("Skip", "1")]
+    [DataRow("SkipWhile", "x => true")]
+    [DataRow("Join", "[1], x => x, x => x, (x, y) => x")]
+    [DataRow("GroupJoin", "[1], x => x, x => x, (x, y) => x")]
+    [DataRow("Concat", "[]")]
+    [DataRow("Zip", "[1], (x, y) => x")]
+    [DataRow("Distinct")]
+    [DataRow("Union", "[]")]
+    [DataRow("Intersect", "[]")]
+    [DataRow("Except", "[]")]
+    [DataRow("Reverse")]
+    [DataRow("SequenceEqual", "[]")]
+    [DataRow("ToArray")]
+    [DataRow("ToList")]
+    [DataRow("ToDictionary", "x => x, x => x")]
+    [DataRow("ToHashSet")]
+    [DataRow("DefaultIfEmpty")]
+    [DataRow("First")]
+    [DataRow("FirstOrDefault")]
+    [DataRow("Last")]
+    [DataRow("LastOrDefault")]
+    [DataRow("Single")]
+    [DataRow("SingleOrDefault")]
+    [DataRow("ElementAt", "1")]
+    [DataRow("ElementAtOrDefault", "1")]
+    [DataRow("Any")]
+    [DataRow("Count")]
+    [DataRow("LongCount")]
+    [DataRow("Contains", "1")]
+    [DataRow("Aggregate", "(x, y) => x")]
+    [DataRow("Sum")]
+    [DataRow("Min")]
+    [DataRow("Max")]
+    [DataRow("Average")]
+    [DataRow("Append", "1")]
+    [DataRow("Prepend", "1")]
+    public void Invocation_SetsNotNull_OnLinqMethodsNullChecking(string method, string args = null)
+    {
+        var code = $$"""
+            using System;
+            using System.Linq;
+            using System.Collections.Generic;
+            public class Sample
+            {
+                public void Main(IEnumerable<int> collection)
+                {
+                    Tag("Before1", collection);
+                    _ = collection.{{method}}({{args}});                                                 // Call as extension
+                    Tag("After1", collection);
+
+                    collection = Untracked();
+                    Tag("Before2", collection);
+                    _ = Enumerable.{{method}}(collection{{(args is null ? string.Empty : "," + args)}}); // Call as normal static method
+                    Tag("After2", collection);
+                }
+
+                private IEnumerable<int> Untracked() => [ 1, 2, 3 ];
+
+                private static void Tag(string name, object arg) { }
+            }
+            """;
+        var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
```

Comment:
This should use `SETestContext.CreateCSMethod` to make the scaffolding smaller. (remove class and anything outer to that class).

You could also use `SETestContext.CreateCS` directly, if you make `Untracked` a local function

Reply:
Same below

Reply:
I tried that at the beginning, but it didn't work because `CreateCSMethod`, while helping with not redefining the `Tag` and the class, it does it at the cost of a rigid class structure: it injects the provided code into a class, with `using System` only.
I need to import `System.Collections.Generic` and `System.Linq`, and I can't import those into the class created by `CreateCSMethod` (I get `CS1529 Line: 5: A using clause must precede all other elements defined in the namespace except extern alias declarations`).